### PR TITLE
Add spent addresses test

### PR DESCRIPTION
**Pull request by DyrellC**
# Description
Add a regression test for spent addresses. This test ensures that a spent addresses are registered as spent from in three scenarios: 
1) As an unconfirmed transaction
2) As a confirmed transaction 
3) After the transaction has been pruned from the db

Finishes #1442  

## Type of change
- Enhancement (a non-breaking change which adds functionality)

# How Has This Been Tested?
Tests run on local nodes and on a minkube setup. 

# Checklist:
- [x] My code follows the style guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by GalRogozinski** [comment]
Waiting to merge #1505 before

**Comment by GalRogozinski** [comment]
apparently there are conflicts @DyrellC

File: python-regression/tests/features/machine6/6_local_snapshots_tests.feature [hunk 1]
```diff
@@ -84,4 +84,50 @@ Feature: Test Bootstrapping With LS
...
+
+    # Check that addresses were spent after transaction have been pruned
+    And "wereAddressesSpentFrom" is called on "nodeE" with:
+      |keys                       |values                   |type             |
+      |addresses                  |LS_SPENT_ADDRESSES       |staticValue      |
+
+    Then the response for "wereAddressesSpentFrom" should return with:
+      |keys                       |values                   |type             |
+      |addresses                  |True                     |boolList         |
+
```
Comment (GalRogozinski) [inline python-regression/tests/features/machine6/6_local_snapshots_tests.feature]: Change Request:
Can you try to fetch the tx from the db and fail if you succeed?
This is to raise confidence that pruning did occur

File: python-regression/util/milestone_logic/milestones.py [hunk 2]
```diff
@@ -38,3 +38,12 @@ def issue_milestone(address, api, index, *reference_transaction):
...
     return milestone
+
+
+def update_latest_milestone(config, node, milestone):
+    if 'latestMilestone' not in config:
+        config['latestMilestone'] = {}
+
+    milestone_hash = Transaction.from_tryte_string(milestone['trytes'][0]).hash
+    milestone_hash2 = Transaction.from_tryte_string(milestone['trytes'][1]).hash
+    config['latestMilestone'][node] = [milestone_hash, milestone_hash2]
```
Comment (GalRogozinski) [inline python-regression/util/milestone_logic/milestones.py]: Question:
I am just a bit confused on what you are doing here.
Is this a global variable?
Shouldn't it be in `world` and not `config`?

If I am correct please change.
If I am misunderstanding something please enlighten me :-)

Comment (DyrellC) [reply python-regression/util/milestone_logic/milestones.py]: Yes it is storing it in the world global. The config value is the world.config that's been passed to it from higher in the code. This updates the latest milestone in the world environment.

**Comment by GalRogozinski** [comment]
Fails with `BadApiResponse: 400 response from node: This operation cannot be executed: The subtangle has not been updated yet.`

**Comment by GalRogozinski** [comment]
I see the test is passing...

I still had one request for a change in the test that wasn't yet completed. Besides that, all is well.

File: python-regression/util/static_vals.py [hunk 3]
```diff
@@ -121,3 +122,49 @@
...
+99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999\
+99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999\
+99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999\
+99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999\
+99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999\
+99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999\
+99999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999999\
+999999999999999999999999999999999999999999999999999999999999999999999999999"
+
+NULL_TRANSACTION_LIST = [TryteString(NULL_TRANSACTION), TryteString(NULL_TRANSACTION), TryteString(NULL_TRANSACTION)]
```
Comment (GalRogozinski) [inline python-regression/util/static_vals.py]: Codacy Issue found: ['TryteString' may be undefined, or defined from star imports: iota](https://app.codacy.com/app/GalRogozinski/iri/pullRequest?prid=3727297)

File: python-regression/tests/features/machine6/6_local_snapshots_tests.feature (hunk 1 above)
Comment (GalRogozinski) [reply python-regression/tests/features/machine6/6_local_snapshots_tests.feature]: @DyrellC 
I saw you pushed but didn't see you address this

If you think this shouldn't be done write a comment (I want this though, so I'll be pushy)

**Comment by DyrellC** [comment]
@galrogo  Another buildkite failure that wasn't due to the tests. I've contacted Sadjy already regarding the service being down. Hopefully it's resolved by tomorrow and I can get this and 1584 passed via buildkite again as well

File: python-regression/tests/features/machine6/6_local_snapshots_tests.feature (hunk 1 above)
Comment (DyrellC) [reply python-regression/tests/features/machine6/6_local_snapshots_tests.feature]: I did address this. I added extra steps to the gherkin file just below

Comment (DyrellC) [reply python-regression/tests/features/machine6/6_local_snapshots_tests.feature]: @galrogo

**Pull request merged**